File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        
        count=0
        occurence=len(nums)/2
        
        if(len(nums)==1):
            return nums[0]
        
        distinct=set(nums)
        dup=list(distinct)
        
        for i in dup:
            count=nums.count(i)
            if(count>occurence):
                return i
            else:
                count=0
            
                

File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0, majority;
        for (int num : nums) {
            if (!count) {
                majority = num;
            }
            count += num == majority ? 1 : -1;
        }
        return majority;
    }
};